pkg/render/logstorage.go

```diff
@@ -330,7 +341,7 @@ func (es *elasticsearchComponent) Objects() ([]runtime.Object, []runtime.Object)
 		)
 	}
 
-	if es.dexCfg != nil {
+	if es.dexCfg != nil && es.elasticLicenseType != ElasticLicenseTypeBasic {
```

Comment:
To be clear about what would happen here if elasticLicenseType was Unknown and then went to TypeBasic (and dex was configured).
While the type was unknown we would be configuring dex integration. Would ES just ignore the config or it would error with the config provided?
Once the type was available and Basic the configuration would be updated and ES would run normal.

Reply:
> While the type was unknown we would be configuring dex integration. Would ES just ignore the config or it would error with the config provided?

Ideally it should never be unknown as we are reading license values written by ECK operator, if ECK adds support to new license type we will have to catch that during ECK version update and change this condition. Among the licenses that ECK now supports, basic is only type that doesn't support SSO.
If license type is unknown and dex is configured, and say if ES doesn't support OIDC then, ES will work but it will log errors about OIDC and also Kibana will throw license related error.

Reply:
Would it be better to ensureLicenseType is Enterprise or EnterpriseTrial? Then we'll be sure we have read the value and won't ever configure Dex without it being a type that supports OIDC?
Is there a downside to do that?

Reply:
yes, I've updated to check for Enterprise or EnterpriseTrial.